### ISSUE TITLE
Request: a getSession method?

### ISSUE THREAD

**Issue by j3gb3rt** [expressjs/session#322]
I'm working with an application that has other means of communication the do not go through the middleware. I have managed to do a hack job of unsigning cookies on the outside requests, but it would be nice to have a method as part of the middleware to take either a cookie string or a request object and return the session data.

Hope you'll consider the request

**Comment by dougwilson** [comment]
Use the `.get` method on the `store` you are using. That's what this module does internally.

**Comment by j3gb3rt** [comment]
Right. That's what I had to do after a bunch of cookie parsing. The store's get method takes an Id and in order to get the id, I had to parse then unsign the cookie. I was hoping for a convenience method that would not require me to save my private key that I pass to express-session so that my call to unsign can use it.

**Comment by dougwilson** [comment]
Oops, sorry, @j3gb3rt, I misunderstood your request!

**Comment by ejwaibel** [comment]
👍 

Can you just export the `store` object that is created so that we could access it from `session.store` and allow the ability to use all the methods associated with the store.

**Comment by kennycastro007** [comment]
Is this issue still open? @dougwilson

**Comment by knoxcard** [comment]
@ejwaibel - i need your feature request added now!  I've made a workaround, but it looks ugly as sin!

```
   var redis_store = new (require('connect-redis')(express_session))()
  app.set('redis_store', redis_store)
 // instantiate express-session
```


`app.get('redis_store').destroy()
`

**Comment by knoxcard** [comment]
Check out this solution! I am rolling with this...

**Comment by korobaka** [comment]
no need to do some complicated things! you can just create a middleware like this:

index.js
```
app.use(require('./middlewares/sessions.js'));
```
middlewares/sessions.js
```
module.exports = (req, res, next) => {
    if (!req.app.sessions) req.app.sessions = {};
    req.app.sessions[req.session.id] = req.session;
    next();
}
```
you can now log all your sessions by doing `app.sessions[sessionid]`!
